lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -158,6 +160,16 @@ def warn_about_bare_error
                         "`RSpec::Expectations.configuration.warn_about_potential_false_positives = false`")
         end
 
+        def warn_about_negative_false_positive(type)
+          RSpec.warning("Using `#{type}` risks false positives, since `#{type}` " \
+                       "won't match when Ruby raises a different error (e.g. " \
+                       "`NoMethodError`, `NameError` or `ArgumentError`), potentially " \
+                        "allowing the expectation to pass without even executing the " \
+                        "method you are intending to call. Instead consider using " \
```

Comment:
I think the language about "won't match when Ruby raises a different error" is a little confusing.  How about this instead?

> Using `#{expression}` risks false positives, since literally _any other error_ would cause the
> expectation to pass, including those raised by Ruby (e.g. `NoMethodError`, `NameError` and `ArgumentError`),
> meaning the code you are intending to test may not even get reached.  Instead consider... (the rest of your message)